### ISSUE TITLE
LM-4201: Demo for DNA Explained - add instructions

### ISSUE THREAD

**Issue by austinMarcos** [CiscoDevNet/devnet-express-dna-issues#10]
Add instructions to the story to specify what the event host should be showing on their screen (Spark message, etc.).

**Comment by austinMarcos** [comment]
Armen has re-written the lab instructions to clarify how to launch the script that hosts will use in an event.

**Comment by austinMarcos** [comment]
Armen - the demo has been transferred to container. More work being done on the instructions for the hosts to use the demo. Other minor issues being worked on include adding code that will cause bot to send message with the bot's email address. 

Chris L. is in the loop through Spark room chats.  Armen will have give estimate on Monday for QA to begin. Yawming to start testing week of April 24th.

**Comment by yawming** [comment]
the demo has been transferred to container. More work being done on the instructions for the hosts to use the demo.  - where is it ?

**Comment by RuthCattell** [comment]
Marcos sent an email yesterday to say 
"Armen-

When ready, please upload the Word doc with the instructions for the hosts to use the demo:
https://cisco.app.box.com/files/0/f/22683757875/C_-_proposed_revisions_for_slides_post-v2"

Can someone confirm what's happening with this please?  Has it gone live?  Is it ready?  There's notning in that box folder relating to this.

**Comment by armartirosyan** [comment]
The final instruction document has been provided to Yawming for QA. 
The document has been uploaded to the https://cisco.app.box.com/files/0/f/22683757875/C_-_proposed_revisions_for_slides_post-v2 folder

Awaiting for Yawming to provide a feedback on the QA.

**Comment by yawming** [comment]
Hi Armen,

I was trying to verify the content but as in the email I sent last week indicated I cannot find a dClund environment matched to the content.

This instruction is only for instructor or both student and instructor?

If it’s for both then the first I think we should be more detail to tell where they can obtain the login and password.

Any way If possible I like finish reviewing this instruction with the environment that specified in it. For example :

“.      In terminal navigate to ~/Code/devnet-express-code-samples/LM-4201 folder”, then I like to know where is the repo for this?
If it’s irrelevant please let me know.
I will finish it  by end of 5/2.


Thanks,

Yawming
-----------------------------

Where is the new dcloud environment for the “Instructions for VIRL Demo in Ubuntu VM” ?

I am using “Cisco DevNet Express DNA Sandbox v2” and still seeing modulexx folder but not LB-xxxx folder ?

1.      In terminal navigate to ~/Code/devnet-express-code-samples/LM-4201 folder. Check the contents of the folder with ll command. You should see 2 files and one folder:
·         sims (Folder) – holds virl topologies
·         demo_bot.py (python script) – main python script which starts the bot and creates webhooks
·         virl.py (python script) – additional python script which is imported into demo_bot.py


Thanks,

Yawming

**Comment by armartirosyan** [comment]
Hi Yawming,

The instructions are only for instructors/presenters. 

To reserve a pod search for Cisco DevNet Express DNA Sandbox v2 in the dCloud catalog  or follow this link [https://dcloud2-sjc.cisco.com/content/demo/2032?returnPathTitleKey=content-view](https://dcloud2-sjc.cisco.com/content/demo/2032?returnPathTitleKey=content-view). 

Note that you can only use your Cisco account to reserve a pod.

Please let me know if you have any questions.

**Comment by yawming** [comment]
I don’t have issue to reserved dCould just that the content I saw it’s not the same ae the one in instruction.

https://dcloud2-sjc.cisco.com/session/13405/details?returnPathTitleKey=view-session


[cid:image001.png@01D2C3F2.EE5CB240]

**Comment by austinMarcos** [comment]
I updated the instructions, and renamed the filename to:
**Instructions for LM-4201 Demo.docx**

Make sure to make any changes to the latest doc in Box:

https://cisco.app.box.com/files/0/f/22683757875/C_-_proposed_revisions_for_slides_post-v2

**Comment by yawming** [comment]
Instructions for VIRL Demo in Ubuntu VM:


1.      After getting into session  - https://dcloud2-sjc.cisco.com/content/demo/2032?returnPathTitleKey=content-view. Still cannot find the ~/Code/devnet-express-code-samples/LM-4201

Folder.

Instructions for VIRL Demo in Docker Container:


1.      Maybe have a pointer for people don’t have Docker in their system to tell them how to install it?

2.      I was running Docker on my Mac, was able to follow the instruction until –

“Run [attached virl file] – bot will download the attached.virl file and run it on VIRL server”

Don’t know where to get virl_demo.virl. Maybe have more detail for this step ?

**Comment by armartirosyan** [comment]
Hi Yawming,

Can you please try these commands in the terminal and see if it pulls necessary folders from github
```bash
cd ~/Code/devnet-express-code-samples/
git fetch --all
git reset --hard origin/master
```
I will add a note to instructions so the presenters can pull the repo and access the folder.

Regarding the `run [attached virl file]` command in the ~/Code/devnet-express-code-samples/LM-4201/sims folder you can find a .virl file and test the command against it. I will add a note for that as well.

Thanks

**Comment by armartirosyan** [comment]
I found out why the git pull was not working.  Apparently the Ubuntu box was imaged under `iss47` branch and that is why Yawming was not able to see the necessary folders. We need to ask Eddie to re-image the Ubuntu box.

**Comment by armartirosyan** [comment]
As a temporary fix I updated iss47 branch with the contents currently in the master branch.

**Comment by austinMarcos** [comment]
Fixed with the re-image of pod v2.1.
